epub_translator/epub2/html/file.py

```
@@ -0,0 +1,81 @@
+
+_FILE_HEAD_PATTERN = re.compile(r"^<\?xml.*?\?>[\s]*<!DOCTYPE.*?>")
+_XMLNS_IN_TAG = re.compile(r"\{[^}]+\}")
+_BRACES = re.compile(r"(\{|\})")
+
+
+class HTMLFile:
+    def __init__(self, file_content: str):
+        match = re.match(_FILE_HEAD_PATTERN, file_content)
+        xml_content = re.sub(_FILE_HEAD_PATTERN, "", to_xml(file_content))
+        self._head: str = match.group() if match else None
+        self._root: Element = fromstring(xml_content)
```

Ruff (0.14.10)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)